File: src/game/deck.js
// カードのスーツとランク定義
export const SUITS = ['♠', '♥', '♦', '♣'];
export const SUIT_COLORS = { '♠': 'spade', '♥': 'heart', '♦': 'diamond', '♣': 'club' };
export const RANKS = ['2', '3', '4', '5', '6', '7', '8', '9', '10', 'J', 'Q', 'K', 'A'];
export const RANK_VALUES = {
  '2': 2, '3': 3, '4': 4, '5': 5, '6': 6, '7': 7, '8': 8,
  '9': 9, '10': 10, 'J': 11, 'Q': 12, 'K': 13, 'A': 14
};

/**
 * 52枚のデッキを生成
 */
export function createDeck() {
  const deck = [];
  for (const suit of SUITS) {
    for (const rank of RANKS) {
      deck.push({ suit, rank, id: `${rank}${suit}` });
    }
  }
  return deck;
}

/**
 * Fisher-Yates シャッフル
 */
export function shuffleDeck(deck) {
  const shuffled = [...deck];
  for (let i = shuffled.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [shuffled[i], shuffled[j]] = [shuffled[j], shuffled[i]];
  }
  return shuffled;
}

/**
 * デッキからカードを配布
 */
export function dealCards(deck, count) {
  const dealt = deck.slice(0, count);
  const remaining = deck.slice(count);
  return { dealt, remaining };
}

File: backend/game/logic/deck.py
"""デッキの生成・シャッフル・配布"""
import random

SUITS = ['♠', '♥', '♦', '♣']
SUIT_COLORS = {'♠': 'black', '♥': 'red', '♦': 'red', '♣': 'black'}
RANKS = ['2', '3', '4', '5', '6', '7', '8', '9', '10', 'J', 'Q', 'K', 'A']
RANK_VALUES = {
    '2': 2, '3': 3, '4': 4, '5': 5, '6': 6, '7': 7, '8': 8,
    '9': 9, '10': 10, 'J': 11, 'Q': 12, 'K': 13, 'A': 14
}


def create_deck():
    """52枚のデッキを生成"""
    return [{'suit': suit, 'rank': rank, 'id': f'{rank}{suit}'}
            for suit in SUITS for rank in RANKS]


def shuffle_deck(deck):
    """デッキをシャッフル"""
    shuffled = deck.copy()
    random.shuffle(shuffled)
    return shuffled


def deal_cards(deck, count):
    """デッキからカードを配布"""
    return {
        'dealt': deck[:count],
        'remaining': deck[count:]
    }
